### PR TITLE
[OPENJDK-123] pkg-update module

### PR DESCRIPTION
**Pull request by jmtd**
Performs a package update (yum, or dnf, or microdnf)

### CODE REVIEW

File: jboss/container/util/pkg-update/execute.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+        mgr="/usr/bin/$mgr"
+        break
+    fi
+done
+if ! test -x "$mgr"; then
+    echo "cannot find a package manager" >&2
+    exit 1
+fi
+
+"$mgr" update -y
```
Comment (jerboaa) [inline jboss/container/util/pkg-update/execute.sh:15]: Isn't this missing a `[..] && $mgr -y clean all` or some such?

File: jboss/container/util/pkg-update/execute.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+set -euo pipefail
+
+for mgr in yum dnf microdnf; do
+    if test -x "/usr/bin/$mgr"; then
```
Comment (jerboaa) [inline jboss/container/util/pkg-update/execute.sh:5]: I'd suggest to use `command -v "$mgr"` here over `test -x` with a path. This avoids hard-coding `/usr/bin`

File: jboss/container/util/pkg-update/execute.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        mgr="/usr/bin/$mgr"
```
Comment (jerboaa) [inline jboss/container/util/pkg-update/execute.sh:6]: Perhaps `mgr=$(command -v $mgr)` here?

File: jboss/container/util/pkg-update/execute.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        break
+    fi
+done
+if ! test -x "$mgr"; then
```
Comment (jerboaa) [inline jboss/container/util/pkg-update/execute.sh:10]: Have you considered initializing `mgr` to say, `not-exist`, and test against that here? Like `if [ "$mgr" == "not-exist"]; then ...`

File: jboss/container/util/pkg-update/module.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+schema_version: 1
+name: jboss.container.util.pkg-update
+version: '1.0'
+description: "Perform a package manager update."
```
Comment (jerboaa) [inline jboss/container/util/pkg-update/module.yaml:4]: s/Perform a package manager update/Perform a system update via the package manager/

**Review by jerboaa — CHANGES_REQUESTED** [review]
A couple of comments. Let me know what you think.

**Comment by jmtd** [comment]
@jerboaa Thank you for the review! I've pushed a commit that addresses your points, please let me know what you think.

File: jboss/container/util/pkg-update/execute.sh [hunk 6]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -euo pipefail
+
+for candidate in yum dnf microdnf; do
+    if command -v "$candidate"; then
+        mgr="$(command -v "$candidate")"
+        "$mgr" update -y
+        "$mgr" -y clean all
```
Comment (jerboaa) [inline jboss/container/util/pkg-update/execute.sh:8]: Should we combine these lines? The intention for this is to clean files which got cached due to the `$mgr update -y` command, right? I.e. `"$mgr" update -y && "$mgr" -y clean all`

Comment (jmtd) [reply jboss/container/util/pkg-update/execute.sh:8]: The whole script gets executed in one container layer so there's no need to do that

Comment (jerboaa) [reply jboss/container/util/pkg-update/execute.sh:8]: OK.

**Review by jerboaa — APPROVED** [review]

**Pull request merged**